src/dispatch/EnhancedAuthentication.ts

```diff
@@ -87,8 +87,10 @@ export class EnhancedAuthentication {
                 // The expiration property of Credentials has a date type. Because the date was serialized as a string,
                 // we need to convert it back into a date, otherwise the AWS SDK signing middleware
                 // (@aws-sdk/middleware-signing) will throw an exception and no credentials will be returned.
-                credentials.expiration = new Date(credentials.expiration);
-                this.credentials = credentials;
+                this.credentials = {
+                    ...credentials,
+                    expiration: new Date(credentials.expiration as Date)
+                };
```

Comment:
issue: We need unit tests to cover the new behavior (here and in Authentication.ts).

I think this change introduces a bug. Cover the following cases:
1. When credentials are read from storage.
2. When credentials are read from storage, and then read from the member variable.